src/xmlutil.rs

```diff
@@ -250,4 +282,23 @@ mod tests {
         }
     }
 
+    #[test]
+    fn skip_tree_with_subtree() {
+        let file = File::open("tests/sample-data/skip_tree.xml").unwrap();
+        let file = BufReader::new(file);
+        let mut my_parser  = EventReader::new(file);
+        let my_stack = my_parser.events().peekable();
+        let mut reader = XmlResponseFromFile::new(my_stack);
+
+        // skip two leading fields since we ignore them (xml declaration, return type declaration)
+        reader.next();
+        reader.next();
```

Comment:
I might be wrong about this, but the code I wrote [here](https://github.com/rusoto/rusoto/pull/235/files#diff-ff8d2313f22100cddc2afca5c3e20414R150) might be relevant to this problem.

I can't remember it exactly now but I think I had the same problem and I wrote that to work around it.